lib/audit_help.c

```diff
@@ -63,8 +81,17 @@ void audit_logger (int type, MAYBE_UNUSED const char *pgname, const char *op,
 	if (audit_fd < 0) {
 		return;
 	} else {
-		audit_log_acct_message (audit_fd, type, NULL, op, name, id,
-		                        NULL, NULL, NULL, result);
+		/*
+		 * The audit system needs white space in the op field to
+		 * be replaced with dashes so that parsers get the whole
+		 * field. Not all C libraries have strdupa.
+		 */
+		char *tmp_op = alloca (strlen (op) + 1);
```

Comment:
strlen() + 1 doesn't enforce a string literal.

Using alloca(3) on whatever we pass is dangerous.  Please add a macro that validates the input and calls alloca().

Reply:
Yeah, but we might forget and change some of those in the future.  That's why I wanted a STRDUPA() macro that enforces that the input is a string literal.